test/libyul/ewasmTranslationTests/balance.yul

```diff
@@ -6,7 +6,7 @@
 // Trace:
 // Memory dump:
 //      0: 0000000000000000000000000000000000000000000000000000000000000001
-//     20: 0000000000000000000000000000000000000000000000000000000022222222
+//     20: 0000000000000000000000000000000022222222000000000000000000000000
 // Storage dump:
-//   0000000000000000000000000000000000000000000000000000000000000000: 0000000000000000000000000000000000000000000000000000000022222222
-//   0000000000000000000000000000000000000000000000000000000000000001: 0000000000000000000000000000000000000000000000000000000022222222
+//   0000000000000000000000000000000000000000000000000000000000000000: 0000000000000000000000000000000022222222000000000000000000000000
+//   0000000000000000000000000000000000000000000000000000000000000001: 0000000000000000000000000000000022222222000000000000000000000000
```

Comment:
Are you sure this is correct? The previous byte ordering seems to be correct.

Reply:
hmm.. that looks strange

Reply:
@axic @chriseth I'm not sure what is exactly going on here. I'm either able to let the semantic test `test/libsolidity/semanticTests/various/balance.sol` pass, or the ewasm translation test `test/libyul/ewasmTranslationTests/address.yul`. But one thing that I noticed - if I just compare the hera implementation with the helper functions defined in `EwasmBuiltinInterpreter` - is that in hera addresses are stored with `storeMemory(..)`, where e.g. `storeUint128(..)` uses `storeMemoryReverse(..)`, but in the  `EwasmBuiltinInterpreter` these functions are defined as
```
void writeU128(uint64_t _offset, u256 _value) { writeU256(_offset, std::move(_value), 16); }
void writeAddress(uint64_t _offset, u256 _value) { writeU256(_offset, std::move(_value), 20); }
```
where in both cases `writeU256` is used. In `EwasmBuiltinInterpreter` `address` and `u128` use the same functions, in contrast to hera. It looks like that either hera or the `EwasmBuiltinInterpreter` is wrong here. Maybe I miss something. Any ideas?

Reply:
See [the data types](https://github.com/ewasm/design/blob/master/eth_interface.md#data-types). `u128` is considered as a number and stored as little endian, whereas `address` is considered a bytearray and hence just copied.

So basically I guess `EwasmBuiltinInterpreter` is wrong.

Reply:
Great! I will look into that!